File: HelloToast2/app/src/main/java/com/example/hellotoast/MainActivity.java
package com.example.hellotoast;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
    private Button btnToast, btnCount;
    private TextView textToast;
    private int count = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        btnToast = findViewById(R.id.btnToast);
        btnCount = findViewById(R.id.btnCount);
        textToast = findViewById(R.id.textToast);

        // Xử lý xự kiện
        btnToast.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showToast(v);
            }
        });
        btnCount.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                countUp(v);
            }
        });
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
    public void showToast(View v){
        Toast.makeText(MainActivity.this, R.string.toast_message, Toast.LENGTH_LONG).show();
    }
    public void countUp(View v){
        count++;
        textToast.setText(String.valueOf(count));
    }
}

File: AutoSMSPhone/app/src/main/java/com/example/autosmsphone/MainActivity.kt
package com.example.autosmsphone

import android.Manifest
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat

class MainActivity : AppCompatActivity() {
    private val REQUEST_PERMISSIONS = 1
    private lateinit var txtStatus: TextView
    private lateinit var btnToggle: Button
    private lateinit var sharedPreferences: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        txtStatus = findViewById(R.id.txtStatus)
        btnToggle = findViewById(R.id.btnToggle)
        sharedPreferences = getSharedPreferences("AutoReplyPrefs", Context.MODE_PRIVATE)

        checkPermissions()
        updateUI()

        btnToggle.setOnClickListener {
            toggleAutoReply()
        }
    }

    private fun checkPermissions() {
        val permissions = arrayOf(
            Manifest.permission.READ_PHONE_STATE,
            Manifest.permission.SEND_SMS
        )

        val permissionsNeeded = permissions.filter {
            ContextCompat.checkSelfPermission(this, it) != PackageManager.PERMISSION_GRANTED
        }

        if (permissionsNeeded.isNotEmpty()) {
            ActivityCompat.requestPermissions(this, permissionsNeeded.toTypedArray(), REQUEST_PERMISSIONS)
        } else {
            Toast.makeText(this, "Tất cả quyền đã được cấp!", Toast.LENGTH_SHORT).show()
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == REQUEST_PERMISSIONS) {
            if (grantResults.all { it == PackageManager.PERMISSION_GRANTED }) {
                Toast.makeText(this, "Quyền đã được cấp!", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "Ứng dụng cần quyền để hoạt động!", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun toggleAutoReply() {
        val isEnabled = sharedPreferences.getBoolean("autoReplyEnabled", false)
        val newState = !isEnabled

        sharedPreferences.edit().putBoolean("autoReplyEnabled", newState).apply()
        updateUI()

        val message = if (newState) "Chế độ tự động trả lời đã BẬT!" else "Chế độ tự động trả lời đã TẮT!"
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
    }

    private fun updateUI() {
        val isEnabled = sharedPreferences.getBoolean("autoReplyEnabled", false)
        txtStatus.text = if (isEnabled) "Chế độ tự động trả lời: Bật" else "Chế độ tự động trả lời: Tắt"
        btnToggle.text = if (isEnabled) "Tắt chế độ tự động trả lời" else "Bật chế độ tự động trả lời"
    }
}
    